src/metrickit.cs

```diff
@@ -0,0 +1,356 @@
+//
+// MetricKit C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+using System;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace MetricKit {
+
+	interface NSUnitDuration : NSUnit { }
+	interface NSUnitInformationStorage : NSUnit { }
+
+	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	interface MXMetric : NSSecureCoding {
+		[Export ("JSONRepresentation")]
+		NSData JsonRepresentation { get; }
+
+		[Export ("DictionaryRepresentation")]
+		NSDictionary DictionaryRepresentation { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[BaseType (typeof(MXMetric), Name = "MXCPUMetric")]
+	interface MXCpuMetric {
+		[Export ("cumulativeCPUTime", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitDuration> CumulativeCpuTime { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[BaseType (typeof(MXMetric), Name = "MXGPUMetric")]
+	interface MXGpuMetric {
+		[Export ("cumulativeGPUTime", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitDuration> CumulativeGpuTime { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[BaseType (typeof(NSDimension))]
+	[DisableDefaultCtor]
+	interface MXUnitSignalBars : NSUnit {
```

Comment:
why `: NSUnit` ? the base type is set with an attribute and `NSUnit` is not a protocol

Reply:
If I do so, I get the following compilation error:

`
 The type 'MetricKit.MXUnitSignalBars' cannot be used as type parameter 'UnitType' in the generic type or method 'MXHistogram<UnitType>'. There is no implicit reference conversion from 'MetricKit.MXUnitSignalBars' to 'Foundation.NSUnit'.
`

Reply:
The generated code with my code looks good: 

```csharp
[Register("MXUnitSignalBars", true)]
        [Unavailable (PlatformName.WatchOS, PlatformArchitecture.All)]
        [Unavailable (PlatformName.TvOS, PlatformArchitecture.All)]
        [Unavailable (PlatformName.MacOSX, PlatformArchitecture.All)]
        [Introduced (PlatformName.iOS, 13,0, PlatformArchitecture.All)]
        public unsafe partial class MXUnitSignalBars : NSDimension, INSCoding, INSCopying, INSSecureCoding {
                
                [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
                static readonly IntPtr class_ptr = Class.GetHandle ("MXUnitSignalBars");
                
                public override IntPtr ClassHandle { get { return class_ptr; } }
                
                [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
```

Reply:
I think this is fine, what is happening is that temp.dll build does no know about inheritance chain yet due to at this point is metadata, by doing `interface MXUnitSignalBars : NSUnit` you create that link but the generator will do the right thing and ignore it. I would still add a comment to the code tho as @spouliot correctly points out `NSUnit` is not a protocol so better remind our future selves why we did this.